common/arch/sdl/joy.cpp

```diff
@@ -466,5 +467,48 @@ int apply_deadzone(int value, int deadzone)
 		return 0;
 }
 
+bool joy_translate_menu_key(const d_event &event) {
+	if (event.type != EVENT_JOYSTICK_BUTTON_DOWN)
+		return false;
+#if DXX_MAX_BUTTONS_PER_JOYSTICK || DXX_MAX_HATS_PER_JOYSTICK || DXX_MAX_AXES_PER_JOYSTICK
```

Comment:
I think this could be simplified to `#if DXX_MAX_JOYSTICKS`.  `SConstruct` ensures that the maximum number of joysticks is forced to `0` if none of the inputs are enabled.

https://github.com/dxx-rebirth/dxx-rebirth/blob/096a678ff759dd19f5a1d50036931926f17944e1/SConstruct#L1423-L1424

Reply:
OK. Again, this was copied from other parts of the code.

Reply:
I will fix the existing versions to eliminate this misleading hint.

Reply:
I see the problem.  A previously contributed improvement blurred the distinction between axes and buttons, and required some preprocessor adjustments to ensure the right variables were present.  Prior to that, the tests would be for (buttons || hats), but not axes.  To support that change, the contributor added `|| axes` in various places, and I did not ask for it to be handled differently then, so now there are constructs which can mislead readers.